### ISSUE TITLE
Invalid Storage Proof given out

### ISSUE THREAD

**Issue by nakul1010** [scs/substrate-api-client#463]
Used the `get_storage_value_proof` method to get proof for custom pallet `OCEX` with key as `IngressMessages` of polkadex chain. It gives out an invalid hash. If the same thing is done using jsonrpsee it provides the correct hash. The problem is only faced in custom  build pallets 

Code used to get hash using substrate API client
```
        let result= self.api
            .get_storage_value_proof("OCEX", "IngressMessages", Some(at))??;// api is of type Api<sr25519::Pair, WsRpcClient, PlainTipExtrinsicParams> 
```

Code used to get hash using jsonrpsee API client
```
        let ingress_messages: Value = client
            .request(
                "state_getReadProof",
                rpc_params![vec![to_value(storage_key.clone()).unwrap()], block_hash.clone()],
            )
            .await
            .unwrap();
            let read_proof: ReadProof<Hash> = serde_json::from_str(&ingress_messages.to_string()).unwrap();
```

**Comment by haerdib** [comment]
Interesting. There should be no difference between the pallets - be it custom or from parity. 
1. Which api client version are you using?
2. Do the storage keys, one generated by the api-client (should be printed as Info log), and the self generated one match?

One possible difference I'm seeing is that the api-client deserializes to an `Result<Option<ReadProof<Hash>>>`, while for code you posted with the jsonrpsee, I don't see any option involved.

But if that would be the problem, I don't see it working for one pallet but not for the other.

**Comment by nakul1010** [comment]
We are using this branch `polkadot-v0.9.27` and both the storage keys are matching.

**Comment by haerdib** [comment]
THB I'm a little stuck. I could test the scenario with common pallets, but if it's working with them, that wouldn't help.

Can you provide me with links / example code on how I can reproduce your issue?

**Comment by masapr** [comment]
@nakul1010 can you provide the full example? I don't see how you build the storage_key for the pure jsonrpsee usage, but I think it is important.
... I'm trying to reproduce the problem with the polkadex/mainnet node from dockerhub (which should be correct (?) ).

**Comment by masapr** [comment]
I gave it a try to reproduce it on the branch `polkadot-v0.9.27` together with the node `polkadex/mainnet:v4.0.0` (from docker hub). But for me the storage proof does not differ.

Do you still have this issue? If so, can you provide the full code, that you're running?

**Comment by nakul1010** [comment]
Hey @haerdib and @masapr,
I'm sorry for not getting back to you sooner. After some investigation, we discovered that the issue was not with the substrate API client method as we initially suspected. Instead, the culprit was the use of the `to_string` method, which introduced extra bytes and caused a cascade of problems throughout our codebase.

Thank you both for your help in troubleshooting this issue. With your assistance, we were able to identify and resolve the problem. Therefore, I am closing the issue.
